### PR TITLE
Update quickstart for v0.5

### PR DESCRIPTION
**Pull request by jbischof**
* Showcase unified KerasCV/KerasNLP API
* Remove `Augmenter`
* Add `jbischof` as author

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,55 +50,65 @@ pip install git+https://github.com/keras-team/keras-cv.git tensorflow --upgrade
 ## Quickstart
 
 ```python
-import keras_cv
 import tensorflow as tf
 from tensorflow import keras
+from keras_cv.layers import RandomFlip, RandAugment, CutMix, Mixup
```
Comment (LukeWood) [inline README.md:55]: maybe from keras_cv import layers as cv_layers?  Typically we don't really like to prescribe Symbol imports as it hurts trace-ability a bit.

File: README.md [hunk 2]
```diff
@@ -49,55 +50,65 @@ pip install git+https://github.com/keras-team/keras-cv.git tensorflow --upgrade
...
     preprocess_data, num_parallel_calls=tf.data.AUTOTUNE).prefetch(
         tf.data.AUTOTUNE)
 
-# Create a model
-densenet = keras_cv.models.DenseNet121(
-    include_rescaling=True,
-    include_top=True,
-    num_classes=3
+# Create a model using a pretrained backbone
+backbone = ResNet50V2Backbone.from_preset("resnet50_v2_imagenet")
```
Comment (LukeWood) [inline README.md:91]: Maybe EfficientNetV2B0Backbone is the most impressive.   Many fewer parameters; super efficient while maintaining strong power.

**Review by LukeWood — COMMENTED** [review]
Minor comments but LGTM overall

File: README.md (hunk 2 above)
Comment (jbischof) [reply README.md:91]: I don't think that's merged yet.

File: README.md [hunk 3]
```diff
@@ -49,55 +50,65 @@ pip install git+https://github.com/keras-team/keras-cv.git tensorflow --upgrade
...
-    layers=[
-        keras_cv.layers.RandomFlip(),
-        keras_cv.layers.RandAugment(value_range=(0, 255)),
-        keras_cv.layers.CutMix(),
-        keras_cv.layers.MixUp()
+    [
+        RandomFlip(),
+        RandAugment(value_range=(0, 255)),
+        CutMix(),
+        MixUp()
```
Comment (innat) [inline README.md:67]: Feels a bit sceptical to use these two heavy transformation sequentially. I wonder if there's any literature explore this area. Otherwise what if showcase the random_choice layer on this two, no?

```python
layers.RandomChoice(
   layers=[CutMix(), MixUp(), FourierMix()]
)
```

**Review by ianstenbit — APPROVED** [review]
LGTM -- thanks Jonathan!

File: README.md (hunk 2 above)
Comment (LukeWood) [reply README.md:91]: Figure we may as well put our best foot forward and show people we're not just fiddling with outdated components; WRT rock paper scissors EfficientNetV2B0 is much lighter weight than ResNet50V2: 

EfficientNetV2B0: 7_200_312 params
ResNet50V2: 25_613_800 params

Just a thought; we can just change it in a follow up PR too as its a one liner - no need to block these updates.

**Review by LukeWood — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -49,55 +50,65 @@ pip install git+https://github.com/keras-team/keras-cv.git tensorflow --upgrade
...
         tf.data.AUTOTUNE)
 
-# Create a model
-densenet = keras_cv.models.DenseNet121(
-    include_rescaling=True,
-    include_top=True,
-    num_classes=3
+# Create a model using a pretrained backbone
+backbone = ResNet50V2Backbone.from_preset("resnet50_v2_imagenet")
+model = ImageClassifier(
```
Comment (innat) [inline README.md:92]: As this read-me file is the first thing a visitor will make impression on keras-cv, it might be needed to consider adding advanced modeling, i.e. object detection (2d, 3d), segmentation. Maybe adding individual readme to corresponding [modeling directories](https://github.com/keras-team/keras-cv/tree/master/keras_cv/models); just a thoughts.

File: README.md [hunk 5]
```diff
@@ -49,55 +50,65 @@ pip install git+https://github.com/keras-team/keras-cv.git tensorflow --upgrade
...
-densenet.compile(
+model.compile(
     loss='categorical_crossentropy',
-    optimizer='adam',
+    optimizer=keras.optimizers.Adam(learning_rate=1e-5),
     metrics=['accuracy']
 )
 
+# Freeze backbone to avoid overfitting to small dataset
+model.backbone.trainable = False
```
Comment (innat) [inline README.md:104]: Doesn't it require to set `model.compile` right after changing the trainable attributes?

File: README.md (hunk 4 above)
Comment (jbischof) [reply README.md:92]: Eventually we will have a "getting started" guide with a tour of the library, but for now I added a link to our developer guides above

File: README.md (hunk 5 above)
Comment (jbischof) [reply README.md:104]: Worked just fine in my colab (looking at `model.summary()`). Will add a gist once I respond to some comments

File: README.md (hunk 3 above)
Comment (jbischof) [reply README.md:67]: Great we're just waiting for #1749 @LukeWood

File: README.md (hunk 5 above)
Comment (innat) [reply README.md:104]: From the [doc](https://keras.io/guides/transfer_learning/)

image

The `model.summary` might have possible bugs, relevant [ticket](https://github.com/keras-team/keras/issues/18092). Mentioning the author of the above doc to give comment on it, cc. @fchollet

Comment (Frightera) [reply README.md:104]: @innat I guess bug happens when you try to freeze the whole model, when there are nested models they behave like a layer so `model.backbone.trainable` may not be an issue (not tested).

Comment (innat) [reply README.md:104]: @Frightera 
Yes, I think so. 

Now, if we don't recompile the model after changing its trainable attributes, what would happen in real? How it would affect the rest of the code? Does the model summary report something useful? 

[similar [feedback](https://github.com/keras-team/keras-io/pull/1314#discussion_r1164088972) cc. @martin-gorner]

Comment (Frightera) [reply README.md:104]: @innat Model summary should show something useful if you freeze and unfreeze layers like this:
```python
for layer in model.layers[0:10]:
    layer.trainable = True
```
But I am not sure if they are tracked properly without calling `compile()`. Needs further testing I guess.

Comment (jbischof) [reply README.md:104]: Thanks this is a great catch @innat! The model does not train very well with a frozen backbone actually 🥶 🥶 🥶

Comment (innat) [reply README.md]: > The model does not train very well with a frozen backbone actually 

I think it's a different thing from what I was concerned about. I wondered about 

```python
# using!
model = keras.Sequentail([backbone, neck, head])
model.compile( ... )
model.backbone.trainable = Fasle

# or should it be this way!
model = keras.Sequentail([backbone, neck, head])
model.backbone.trainable = Fasle
model.compile( ... )
```

Comment (jbischof) [reply README.md]: No, we are agreeing @innat! My previous runs had your first setup and the backbone was actually training. When I switched to the second the model behaved totally differently and I verified that the backbone was actually frozen. So
* only the second approach works
* a frozen backbone doesn't seem to work well (or would need to some fiddling to work well) in my quickstart
